### ISSUE TITLE
API: Ability to fork project

### ISSUE THREAD

**Issue by rajdip-b** [keyshade-xyz/keyshade#187]
## Description

This function will allow any user to fork a project into any of their workspace, where they have permission to create a project. This is directly dependent on #154, whereby only projects with `accessLevel` set to `GLOBAL` will be allowed to fork. Forking a project will copy over all the configurations contained in a project.

### Technical details

- Users should be able to fork a project only if the `accessLevel` is `GLOBAL`.
- Forking will take the following details:
  - `projectId`: ID of the project to be forked
  - `workspaceId`: (Optional) ID of the workspace to which the fork will belong to. Defaults to the user's default workspace. If any other workspace is specified, the user should have `CREATE_PROJECT` authority in the workspace. Otherwise, it should throw an error.
  - `forkName`: (Optional) Name of the forked project. Defaults to the actual project name. Should throw an error if not specified and a project with the same name already exists in the workspace.
- Forking will copy over all the variables, secrets and environments along with the configuration versions into the user's project.
- Making changes to the fork SHOULD NOT reflect changes in the original project.

### Implementation steps

- Get #154 implemented if not already done
- Add the business logic in `project.service.ts`
- Add the controller logic in `project.controller.ts`
- Add relevant test in `project.e2e.spec.ts`

### Helper function

Additionally, for the convenience of client applications, we would also like to add an endpoint in `workspace` module that will fetch all the workspaces where the users have `CREATE_PROJECT` or higher authorities set. 

## Additional context

We have plans to set up an upstream just like GitHub or GitLab does. But this is set for the future, if and when we get the demand.
